lib/logger_json/formatters/datadog_logger.ex

```diff
@@ -0,0 +1,56 @@
+defmodule LoggerJSON.Formatters.DatadogLogger do
+  @moduledoc """
+  DataDog formatter.
+  """
+  import Jason.Helpers, only: [json_map: 1]
+
+  alias LoggerJSON.{FormatterUtils, JasonSafeFormatter}
+
+  @behaviour LoggerJSON.Formatter
+
+  @processed_metadata_keys ~w[pid file line function module application]a
+
+  def format_event(level, msg, ts, md, md_keys) do
+    Map.merge(
+      %{
+        logger: json_map(
+          thread_name: inspect(Keyword.get(md, :pid)),
+          method_name: method_name(md)
+        ),
+        message: IO.chardata_to_string(msg),
+        syslog: json_map(
+          hostname: node_hostname(),
+          severity: Atom.to_string(level),
+          timestamp: FormatterUtils.format_timestamp(ts)
+        )
+      },
+      format_metadata(md, md_keys)
+    )
+  end
+
+  defp format_metadata(md, md_keys) do
+    LoggerJSON.take_metadata(md, md_keys, @processed_metadata_keys)
+    |> JasonSafeFormatter.format()
+    |> FormatterUtils.maybe_put(:error, format_error(md))
+  end
+
+  defp format_error(md) do
+    with %{reason: reason} <- FormatterUtils.format_process_crash(md) do
+      json_map(
+        stack: reason
+      )
+    end
+  end
+
+  defp method_name(metadata) do
+    function = Keyword.get(metadata, :function)
+    module = Keyword.get(metadata, :module)
+
+    FormatterUtils.format_function(module, function)
+  end
+
+  defp node_hostname do
+    {:ok, hostname} = :inet.gethostname()
+    hostname
```

Comment:
This returns a hostname as chartlist which is later encoded to list of integers (check your examples in PR comment). We need to cast the charlist to binary explicitly